src/ssh/ssh_process.cpp

```diff
@@ -116,17 +128,25 @@ std::string mp::SSHProcess::read_stream(StreamType type, int timeout)
     do
     {
         num_bytes = ssh_channel_read_timeout(channel.get(), buffer.data(), buffer.size(), is_std_err, timeout);
+        mpl::log(mpl::Level::debug, category,
+                 fmt::format("{}:{} {}(): num_bytes={}", __FILE__, __LINE__, __FUNCTION__, num_bytes));
         if (num_bytes < 0)
         {
             // Latest libssh now returns an error if the channel has been closed instead of returning 0 bytes
             if (ssh_channel_is_closed(channel.get()))
-                return std::string();
+            {
+                mpl::log(mpl::Level::debug, category,
+                         fmt::format("{}:{} {}(): channel closed", __FILE__, __LINE__, __FUNCTION__));
+                return output.str();
+            }
 
             throw std::runtime_error(fmt::format("error while reading ssh channel for remote process '{}'"
                                                  " - error: {}",
                                                  cmd, num_bytes));
         }
         output.write(buffer.data(), num_bytes);
+        mpl::log(mpl::Level::debug, category,
+                 fmt::format("{}:{} {}(): output = {}", __FILE__, __LINE__, __FUNCTION__, output.str()));
```

Comment:
I'm worried this might be too much, imagine a long-running command through this, do we want to log that much?

Reply:
Yeah, probably too verbose right here.  I'll take that out.